src/components/playlisteditor/playlisteditor.ts

```diff
@@ -0,0 +1,319 @@
+import { pluginManager } from '../pluginManager';
+import { appRouter } from '../router/appRouter';
+import ServerConnections from '../ServerConnections';
+
+import 'elements/emby-button/emby-button';
+import 'elements/emby-input/emby-input';
+import 'elements/emby-button/paper-icon-button-light';
+import 'elements/emby-select/emby-select';
+
+import 'material-design-icons-iconfont';
+import '../formdialog.scss';
+
+interface DialogElement extends HTMLDivElement {
+    submitted?: boolean
+}
+
+interface PlaylistEditorOptions {
+    items: string[],
+    serverId: string,
+    enableAddToPlayQueue?: boolean,
+    defaultValue?: string
+}
+
+let currentServerId: string;
+
+function onSubmit(this: HTMLElement, e: Event) {
+    const panel = dom.parentWithClass(this, 'dialog') as DialogElement | null;
+
+    if (panel) {
+        const playlistId = panel.querySelector<HTMLSelectElement>('#selectPlaylistToAddTo')?.value;
+
+        loading.show();
+
+        if (playlistId) {
+            userSettings.set('playlisteditor-lastplaylistid', playlistId);
+            addToPlaylist(panel, playlistId)
+                ?.catch(err => {
+                    console.error('[PlaylistEditor] Failed to add to playlist %s', playlistId, err);
+                })
+                .finally(loading.hide);
+        } else {
+            createPlaylist(panel)
+                ?.catch(err => {
+                    console.error('[PlaylistEditor] Failed to create playlist', err);
+                })
+                .finally(loading.hide);
+        }
+    } else {
+        console.error('[PlaylistEditor] Dialog element is missing!');
+    }
+
+    e.preventDefault();
+    return false;
+}
+
+function createPlaylist(dlg: DialogElement) {
+    const apiClient = ServerConnections.getApiClient(currentServerId);
+    const api = toApi(apiClient);
+
+    const itemIds = dlg.querySelector<HTMLInputElement>('.fldSelectedItemIds')?.value || '';
+
+    return getPlaylistsApi(api)
+        .createPlaylist({
+            name: dlg.querySelector<HTMLInputElement>('#txtNewPlaylistName')?.value,
+            ids: itemIds.split(','),
+            userId: apiClient.getCurrentUserId()
+        })
+        .then(result => {
+            dlg.submitted = true;
+            dialogHelper.close(dlg);
+
+            redirectToPlaylist(result.data.Id);
+        });
+}
+
+function redirectToPlaylist(id: string | undefined) {
+    appRouter.showItem(id, currentServerId);
+}
+
+function addToPlaylist(dlg: DialogElement, id: string) {
+    const apiClient = ServerConnections.getApiClient(currentServerId);
+    const api = toApi(apiClient);
+    const itemIds = dlg.querySelector<HTMLInputElement>('.fldSelectedItemIds')?.value || '';
+
+    if (id === 'queue') {
+        playbackManager.queue({
+            serverId: currentServerId,
+            ids: itemIds.split(',')
+        });
+        dlg.submitted = true;
+        dialogHelper.close(dlg);
+        return;
```

Comment:
We probably won't hide `loading` in `finally` because we're not returning `Promise` here.